### ISSUE TITLE
Surface "abandoned" state on projects in UI

### ISSUE THREAD

**Issue by ericholscher** [readthedocs/readthedocs.org#3382]
We now have an inactive project policy: http://docs.readthedocs.io/en/latest/abandoned-projects.html

We should:
* Integrate a public facing "inactive" indicator into the RTD dashboard for projects that qualify

**Comment by stsewd** [comment]
I think we can develop this as two features.

## Marking a doc as _inactive/abandoned_

With this a project that meets the policy requirements for abandoned projects gets marked as inactive/abandoned, this is visible for everyone.

And probably we may want to be able to mark a project as _protected_ (on the django admin panel only), so in a special case this project never gets marked as inactive/abandoned. A special case could be: the author requesting this, for historical reasons, our algorithm is wrong or is a false positive.

## Communicating to the users

I have several ideas for this...

We can run a cronjob and send a warning email to the owner of the project if her/his project is marked as abandoned.

A user wants the project name (from the inactive one), she/he fills a form with some details, the owner of the inactive project receives an email with a warning and the request of the other user. The request can be sent no more than once every time and if a user already sent a request and other wants to do the same, she/he needs to wait a time before sending it.

After the owner is warned several times and no action is taken for reactivate the project, the project name gets released. This is done changing the project name to a random one or {user-name}-inactive-{number}?

If the owner wants to release the project name, she/he can change the name (slug) manually or just delete the project and the user who sent the request gets notified.

## Conclusion

I think with this solution all the process gets simplified and is automatic. And also the interaction between the users is minimal (almost none).

**Comment by bansalnitish** [comment]
Hi @stsewd, I would like to work on this! Can you please get me started!

**Comment by ericholscher** [comment]
I think we shouldn't automatically email/remove abandoned projects. I also think showing abandoned projects is lower priority. People who want a namespace are highly motivated, and will email us, so we should focus on executing that part first.

I think we should keep it light weight in terms of implementation so this isn't a lot to maintain. Perhaps just an admin-only button that says "email project owner with our abandoned project email" as a start, which sends an email from a template that we have in the repo.

Here is the email I sent last time as a start:

```
Hi,

We've had a request from one of our users for the project name {name} on Read the Docs. You are the current owner, and we wanted to reach out to you in accordance with our Abandoned Project policy (http://docs.readthedocs.io/en/latest/abandoned-projects.html).

Since this is a fork of the existing owners project, {name}, you must show us the additional work that has been done in order to keep the new project name, that is different than the existing version of the project.

Please reply to this email either allowing or disallowing the transfer of the name on Read the Docs within 6 weeks, otherwise we will take the action of *initiating the transfer to a new owner* by default.

Cheers,
Eric
```

**Comment by ericholscher** [comment]
At the end of the 6 week period, I renamed the project to `{name}-abandoned`, but I'm not 100% sure that's the correct thing to do.

**Comment by ericholscher** [comment]
I guess the primary thing I want to balance is the amount of work to build this, when it doesn't happen very often, with standardizing the execution. We could even just have a template in the docs, and not automate it within the software, as it has only happened a couple times so far.

**Comment by stsewd** [comment]
@ericholscher so suggesting a template for sending an email to support@readthedocs.org would be fine. I like the idea of keeping it simple for now.

**Comment by bansalnitish** [comment]
I am trying something simple for the start as suggested by @stsewd.
 I will submit a PR soon for the same and then we can discuss and make it better !

**Comment by ericholscher** [comment]
Yea, another simple addition would be to have a query that specified "abandoned", and have a small message on the project page. I'm all for keeping things simple to start.

**Comment by bansalnitish** [comment]
Hi @agjohnson, the PR which I have issued contains the points:
- To mark a project as abandoned
- Admin only option to send mail to project owner
- Tests for checking that only an admin is allowed to send email

The PR misses the following points:
- Abandoned project state as a sidebar item on project detail page.
- Changing the project slug to ``{project}-abandoned`` and notifying the user about that.

I want to know whether I should update the PR to match all the missing requirements.

**Comment by agjohnson** [comment]
@bansalnitish This issue is still in a design decision phase, we'll have some more guidance once we've come to some harder conclusions here.

I looked at the PR you started, but it has a few patterns I think we want to avoid:

* Automating `is_abandonded` -- projects that don't build within a year aren't necessarily abandoned, perhaps just unmaintained. We want to act as an archive in this case, not promote transferring the namespace.
* Surfacing sending email as a user feature. Above I discuss making this an admin feature -- that is, a django site admin feature, not part of the RTD UI

Hold off on making changes for now, as the requirements will likely change. I'll have some more specific feedback after that will hopefully clear up some questions.

**Comment by bansalnitish** [comment]
Sure @agjohnson, I will wait for the design decision to be made and then make the appropriate changes :tada:

**Comment by agjohnson** [comment]
I'm going to unblock this. I think the feature we actually need here is:

* Djano site admin functions for sending an email to project owners that their namespace has been requested
* Djano site admin function for marking a project as abandoned (Adds `-abandoned` to slug, marks `Project.is_abandoned`
* Project detail sidebar piece that states project is abandoned.

In this feature, we don't want:

* User to user email
* Automating abandoned state
* Email on the project forms

Unfortunately, I think your PR is mostly invalidated with these changes -- and there doesn't seem to be admin functions (django site admin functions that is) for any of this. So I'll close your current PR, but leave the branch around for reference. I'd suggest starting with a new PR for this work.

**Comment by dojutsu-user** [comment]
I am working on it.
Will submit a PR soon.

**Comment by dojutsu-user** [comment]
It is still valid. There's an open PR for this.

**Comment by dojutsu-user** [comment]
This is still valid bot.
Can I get a review on the attached PR?

**Comment by dojutsu-user** [comment]
Still valid bot.

**Comment by humitos** [comment]
I'm marking this issue as Accepted since what we need to do is described at https://github.com/readthedocs/readthedocs.org/issues/3382#issuecomment-395885311

**Comment by dojutsu-user** [comment]
@humitos 
There already a PR open -- #4995

**Comment by humitos** [comment]
We wrote two different guides about this issue:

* https://docs.readthedocs.io/en/stable/abandoned-projects.html
* https://docs.readthedocs.io/en/stable/unofficial-projects.html

We discussed the approach of showing a UI element on projects but we weren't able to pattern that works in an accurate way. I'm going to close this issue since we don't plan to implement a semi-automated way of doing this in the near future. We can come back if we think this is still relevant and we have a way to implement a good pattern.
